src/reducers/entities/apps.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import {combineReducers} from 'redux';
+
+import {AppsTypes} from 'action_types';
+import {AppBinding, AppsState} from 'types/apps';
+import {GenericAction} from 'types/actions';
+import {validateBindings} from 'utils/apps';
+
+// This file's contents belong to the Apps Framework feature.
+// Apps Framework feature is experimental, and the contents of this file are
+// susceptible to breaking changes without pushing the major version of this package.
+
+export function bindings(state: AppBinding[] = [], action: GenericAction): AppBinding[] {
+    switch (action.type) {
+    case AppsTypes.RECEIVED_APP_BINDINGS: {
+        validateBindings(action.data);
+        return action.data || [];
```

Comment:
I don't think it should have any real effect, but if we are dispatching RECEIVED_APP_BINDINGS often without a data field, then we may cause unnecessary re-renders as this state changes to new copies of an empty array each time.

Reply:
We don't expect the action to have no data. We are just being cautious there. What may happen is that many times the content is the same (either because it is empty, or because the contents are the same).

Is that handled beautifully, or should we somehow check if the content is equal and if it is, return the original state?

Reply:
Very often, it will be the same. Maybe we should do a deep compare and return the preexisting ref if it is equal